frontend/lib/uast/post-parse-checks.cpp

```diff
@@ -1538,6 +1540,21 @@ void Visitor::visit(const Return* node) {
   }
 }
 
+void Visitor::visit(const Select* node) {
+  checkOtherwiseAfterWhens(node);
+}
+
+void Visitor::checkOtherwiseAfterWhens(const Select* sel) {
+  bool seenOtherwise = false;
+  for(int i = 0; i < sel->numWhenStmts(); i++) {
+    auto when = sel->whenStmt(i);
+    if (seenOtherwise && !when->isOtherwise()) {
+      CHPL_REPORT(context_, WhenAfterOtherwise, sel, sel);
+    } 
+    seenOtherwise |= when->isOtherwise();
+  }
```

Comment:
Continuing with my suggestion of including the offending clauses in the error:
```suggestion
  const When* seenOtherwise = nullptr;
  for(int i = 0; i < sel->numWhenStmts(); i++) {
    auto when = sel->whenStmt(i);
    if (seenOtherwise && !when->isOtherwise()) {
      CHPL_REPORT(context_, WhenAfterOtherwise, sel, seenOtherwise, when);
    } 
    if (when->isOtherwise()) seenOtherwise = when;
  }
```